nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py

```diff
@@ -410,15 +518,28 @@ def migrate_site_and_region_data_to_locations(apps, schema_editor):
         for note in notes:
             note.assigned_object_type = location_ct
             site = Site.objects.get(id=note.assigned_object_id)
-            site_loc = Location.objects.get(name=site.name, location_type=site_lt)
-            note.assigned_object_id = site_loc.id
+            note.assigned_object_id = SITE_TO_LOCATION_LOOKUP.get(site).id
         Note.objects.bulk_update(notes, ["assigned_object_type", "assigned_object_id"], 1000)
 
+        object_changes_changed = ObjectChange.objects.filter(changed_object_type=site_ct)
+        for oc in object_changes_changed:
+            oc.changed_object_type = location_ct
+            site = Site.objects.get(id=oc.changed_object_id)
+            oc.changed_object_id = SITE_TO_LOCATION_LOOKUP.get(site).id
+        ObjectChange.objects.bulk_update(object_changes_changed, ["changed_object_type", "changed_object_id"], 1000)
+
+        object_changes_related = ObjectChange.objects.filter(related_object_type=site_ct)
+        for oc in object_changes_related:
+            oc.related_object_type = location_ct
+            site = Site.objects.get(id=oc.related_object_id)
+            oc.related_object_id = SITE_TO_LOCATION_LOOKUP.get(site).id
+        ObjectChange.objects.bulk_update(object_changes_related, ["related_object_type", "related_object_id"], 1000)
+
```

Comment:
This also feels like a ton of repeated code between the Site and Region cases - more helper functions needed?